### ISSUE TITLE
[a11y]: toggletip implementation bugs

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#12922]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

https://react.carbondesignsystem.com/?path=/story/components-toggletip--default

### Automated testing tool and ruleset

n/a

### Description
There are four primary challenges:
1) None of the toggletips should be visible in storybook until activated. Instead, they are all showing before the i icon is triggered
2) When the toggletip is expanded and has a tabstop inside it, tabbing out of the toggletip should collapse the tip. This is not happening in the [playground](https://react.carbondesignsystem.com/iframe.html?args=defaultOpen:true&id=components-toggletip--playground&viewMode=story) or default views
3) If the focus is in the toggletip when the Esc is pressed, the focus should return to the trigger. Instead, it is unclear where it goes. There is no other place that makes sense except the toggle tip, IMO.
4) There is something called a "Toggletip label" that appears _in front of_ the i icon button. AFAIK, this cannot exist and has no place in the component. It is not listed in the anatomy of the component on the Usage page, for example. Maybe it's meant to demonstrate that the toggletip icon may follow the label for another component, but it is very confusing and should be revamped.
![image](https://user-images.githubusercontent.com/14298245/210825706-706dad7b-5827-4974-8aa1-026c249deae7.png)

### WCAG 2.1 Violation

Focus Order

### Reproduction/example

https://react.carbondesignsystem.com/iframe.html?args=defaultOpen:true&id=components-toggletip--playground&viewMode=story

### Steps to reproduce

1. Go to https://react.carbondesignsystem.com/iframe.html?args=defaultOpen:true&id=components-toggletip--playground&viewMode=story
2. Use a keyboard to navigate to it. The tip will already be open although it has yet to be activated (first issue)
3. On an expanded tip with a tabstop inside it, tab into the tip and keep tabbing until the focus leaves the tip. The tip is not collapsed (second issue)
4. On an expanded tip with a tabstop inside it, tab into the tip and press Esc. The trigger will not get focus (third issue)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Thanks for opening this! I'll try to hit each piece individually here:

> None of the toggletips should be visible in storybook until activated. Instead, they are all showing before the i icon is triggered

Yeah this is intentional - we set the `open` prop to `true` by default in the story so the toggletip internal styling is shown for our automated visual snapshot testing.


> When the toggletip is expanded and has a tabstop inside it, tabbing out of the toggletip should collapse the tip. This is not happening in the [playground](https://react.carbondesignsystem.com/iframe.html?args=defaultOpen:true&id=components-toggletip--playground&viewMode=story) or default views

Yes, this needs to be addressed.


> If the focus is in the toggletip when the Esc is pressed, the focus should return to the trigger. Instead, it is unclear where it goes. There is no other place that makes sense except the toggle tip, IMO.

Yes, this needs to be addressed.


> There is something called a "Toggletip label" that appears in front of the i icon button. AFAIK, this cannot exist and has no place in the component. It is not listed in the anatomy of the component on the Usage page, for example. Maybe it's meant to demonstrate that the toggletip icon may follow the label for another component, but it is very confusing and should be revamped.

The [style documentation/spec](https://carbondesignsystem.com/components/toggletip/style) does show a label and specifies it's typography tokens. @carbon-design-system/design Just to confirm, the intention is that Toggletip can be used with ToggletipLabel, FormLabel, or _no label_ (as shown in the usage/anatomy on the site)

One thing we did find additionally is that we need to update ToggletipLabel to be using body01

**Comment by mbgower** [comment]
thanks @tay1orjones 
> Yeah this is intentional - we set the open prop to true by default in the story so the toggletip internal styling is shown for our automated visual snapshot testing.

Maybe it's possible to have one that does this that is ID'ed as doing that for that reason, and the rest behave as expected?

> Just to confirm, the intention is that Toggletip can be used with ToggletipLabel, FormLabel, or no label (as shown in the usage/anatomy on the site)

That's interesting. I can certainly see a situation where the i icon ends up at the end of a form label (thought I think that would be a separate thing outside the label name, right, since it's in the tab order independently? There's not need to have it announced as part of the input?)
However, I'm having difficulty imagining someone putting in something like:
![image](https://user-images.githubusercontent.com/14298245/213547819-edc0609d-5f11-4868-97ca-d0a40c53af47.png)

Seems pretty redundant. Why not just make the button say "info"?
 Here's what that would look like as a ghost button
![image](https://user-images.githubusercontent.com/14298245/213549099-a2d117e4-12f0-41d2-8685-8732dad8d7c9.png)
and as a tertiary
![image](https://user-images.githubusercontent.com/14298245/213549221-33fd4e0e-d7bb-47c4-86ed-c77c4af57791.png)

Not recommending that, but you see where I'm going, right? A button has a built-in label.  A button with a label external to the button is almost a new one.
Like having a B button with the word 'Bold" in front of it.
